packages/vm/src/imports.rs

```diff
@@ -402,8 +410,12 @@ fn do_ed25519_batch_verify<A: BackendApi, S: Storage, Q: Querier>(
     let public_keys = decode_sections(&public_keys);
 
     let result = ed25519_batch_verify(&messages, &signatures, &public_keys);
-    let gas_info =
-        GasInfo::with_cost(GAS_COST_BATCH_VERIFY_ED25519_SIGNATURE * signatures.len() as u64);
+    let gas_cost = if public_keys.len() == 1 {
+        GAS_COST_ED25519_BATCH_VERIFY_ONE_PUBKEY
+    } else {
+        GAS_COST_ED25519_BATCH_VERIFY
+    } * signatures.len() as u64;
+    let gas_info = GasInfo::with_cost(max(gas_cost, GAS_COST_ED25519_VERIFY));
```

Comment:
Nice calculation to handle single and batch costs together

Reply:
Thanks. It took me some iterations to get there, but I think the idea is sound.

What about a maximum? With 256 signatures in a batch, gast cost would be ~5_236_400 for the costliest (different pubkeys) case.

Reply:
This is "wasm gas" and if we /100 or the "sdk gas", we get 26k - 52k sdk gas for this, which is totally reasonable. 52k gas is about how much the full processing of an external Bank::Send (including tx verification and fee deduction). 40k gas is our current "surcharge" for instantiating a new module that is not pinned (maybe we can reduce this now).

These are very reasonable fees for such calculations. If a batch of 256 ed25519 keys can be verified as quickly as 50 secp256k1 (which benchmarks show), then this is a total win for a wasm tendermint light client